Source/RestClient.swift

```diff
@@ -0,0 +1,133 @@
+                callback?(self.token, nil)
+            } else {
+                callback?(nil, error)
+            }
+        }
+    }
+
+    open func refresh(callback: ((_ token: TokenInfo?, _ error: HTTPError?) -> Void)? = nil) {
+        if let token = token {
+            let parameters: [String: String] = [
+                "grant_type": "refresh_token",
+                "refresh_token": token.refresh_token!,
+                "endpoint_id": token.endpoint_id!
+            ]
+            let headers: [String: String] = ["Authorization": basicAuthToken()]
+            postString("/restapi/oauth/token", parameters: parameters, encoding: URLEncoding.default, headers: headers) { string, error in
+                if error == nil {
+                    self.token = TokenInfo(JSONString: string!)
+                    callback?(self.token, nil)
+                } else {
+                    callback?(nil, error)
+                }
+            }
+        }
+    }
+
+    open func revoke(callback: ((Bool) -> Void)? = nil) {
+        if let token = token {
+            let parameters: [String: String] = ["token": token.access_token!]
+            let headers: [String: String] = ["Authorization": basicAuthToken()]
+            postString("/restapi/oauth/revoke", parameters: parameters, encoding: URLEncoding.default, headers: headers) { string, error in
+                callback?(error == nil)
+            }
+        } else {
+            callback?(true)
+        }
+    }
+
+    open func authorizeUri(redirectUri: String, state: String = "") -> String {
+        var urlComponents = URLComponents(string: server)!
+        urlComponents.path = "/restapi/oauth/authorize"
+        urlComponents.queryItems = [
+            URLQueryItem(name: "response_type", value: "code"),
+            URLQueryItem(name: "state", value: state),
+            URLQueryItem(name: "redirect_uri", value: redirectUri),
+            URLQueryItem(name: "client_id", value: appKey),
+        ]
+        return urlComponents.string!
+    }
+
+    open func authorize(authCode: String, redirectUri: String, callback: ((_ token: TokenInfo?, _ error: HTTPError?) -> Void)? = nil) {
+        let parameters: [String: String] = [
+            "grant_type": "authorization_code",
+            "redirect_uri": redirectUri,
+            "code": authCode,
+        ]
+        let headers: [String: String] = ["Authorization": basicAuthToken()]
+        postString("/restapi/oauth/token", parameters: parameters, encoding: URLEncoding.default, headers: headers) { string, error in
+            if error == nil {
+                self.token = TokenInfo(JSONString: string!)
+                callback?(self.token, nil)
+            } else {
+                callback?(nil, error)
+            }
+        }
+    }
+
+    open func restapi(_ _id: String? = "v1.0") -> RestapiPath {
+        return RestapiPath(parent: PathSegment(parent: nil, _id: nil, rc: self), _id: _id)
+    }
+
+}
```

Comment:
I thought about it carefully and it really took me some time. But I still have a concern which prevent me from making further changes to the code:

1. First and most important, Swift Client is not based on swift SDK. So it doesn't make sense to follow SDK reference. 
    1. Client has many more features than Client: Models, URL builder, More encapsulation from OOP perspective. I do think we need a separate **Client reference** so we can maintain consistency among all of our static and client-style libraries. Now I have done both C# and Swift, I am going to draft and reference myself and submit to the team for review.
1. And for the SDK Reference itself (although it's not quite relevant to client-style projects, I still want to talk about it here), it's based on experience of Python/JS/PHP SDKs which are all dynamic languages. I have rich experience with static languages such as C#/Java/Swift and I feel the dynamic way very unnatural. So may I propose some changes to the SDK reference and let's have a discussion?

My overall suggestion is: merge this PR first (because of #1 above) and let's do more investigation( on #1.1 and #2).

Thank you for your time and your suggestions, really appreciate!